nifi-extension-bundles/nifi-box-bundle/nifi-box-processors/src/main/java/org/apache/nifi/processors/box/FetchBoxFilesInFolder.java

```diff
@@ -0,0 +1,244 @@
+
+    public static final Set<Relationship> RELATIONSHIPS = Set.of(
+            REL_SUCCESS,
+            REL_FAILURE,
+            REL_ORIGINAL
+    );
+
+    private static final List<PropertyDescriptor> PROPERTY_DESCRIPTORS = List.of(
+            BoxClientService.BOX_CLIENT_SERVICE,
+            FOLDER_ID,
+            RECURSIVE_SEARCH,
+            MIN_AGE
+    );
+
+    private volatile BoxAPIConnection boxAPIConnection;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTY_DESCRIPTORS;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final BoxClientService boxClientService = context.getProperty(BoxClientService.BOX_CLIENT_SERVICE)
+                .asControllerService(BoxClientService.class);
+        boxAPIConnection = boxClientService.getBoxApiConnection();
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String folderId = context.getProperty(FOLDER_ID).evaluateAttributeExpressions(flowFile).getValue();
+        final Boolean recursive = context.getProperty(RECURSIVE_SEARCH).asBoolean();
+        final Long minAge = context.getProperty(MIN_AGE).asTimePeriod(TimeUnit.MILLISECONDS);
+
+        try {
+            final long startNanos = System.nanoTime();
+            long createdAtMax = Instant.now().toEpochMilli() - minAge;
+            final List<BoxFile.Info> fileInfos = new ArrayList<>();
+
+            listFolder(fileInfos, folderId, recursive, createdAtMax);
+
+            for (final BoxFile.Info fileInfo : fileInfos) {
+                FlowFile outputFlowFile = session.create(flowFile);
+                try {
+                    final Map<String, String> attributes = BoxFileUtils.createAttributeMap(fileInfo);
+                    outputFlowFile = session.putAllAttributes(outputFlowFile, attributes);
+
+                    final long transferMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                    final String boxUrlOfFile = BoxFileUtils.BOX_URL + fileInfo.getID();
+                    session.getProvenanceReporter().receive(outputFlowFile, boxUrlOfFile, transferMillis);
+
+                    session.transfer(outputFlowFile, REL_SUCCESS);
+                } catch (final Exception e) {
+                    getLogger().error("Failed fetching metadata for file with id [{}] from folder [{}]", fileInfo.getID(), folderId, e);
+                    outputFlowFile = session.putAttribute(outputFlowFile, ERROR_MESSAGE, e.getMessage());
+                    outputFlowFile = session.penalize(outputFlowFile);
+                    session.transfer(outputFlowFile, REL_FAILURE);
```

Comment:
Given that the Box API call is done before, do we expect a potential exception here?